cloud/pkg/dynamiccontroller/application/application.go

```diff
@@ -441,22 +455,71 @@ func (c *Center) Process(msg model.Message) {
 	klog.Infof("[metaserver/applicationCenter]successfully to process Application(%+v)", app)
 }
 
+func (c *Center) createAuthClient(app *Application) (authorizationv1client.AuthorizationV1Interface, error) {
+	parts := strings.SplitN(app.Token, " ", 3)
+	if len(parts) < 2 || strings.ToLower(parts[0]) != "bearer" {
+		return nil, fmt.Errorf("invalid request token format or length: %v", len(parts))
+	}
+
+	token := parts[1]
+	c.authConfig.BearerToken = token
+	return authorizationv1client.NewForConfigOrDie(c.authConfig), nil
+}
+
+func (c *Center) authorizeApplication(app *Application, gvr schema.GroupVersionResource, namespace string, name string) error {
+	tmpAuthClient, err := c.createAuthClient(app)
+	if err != nil {
+		return err
+	}
+	sar := &authorizationv1.SelfSubjectAccessReview{
+		Spec: authorizationv1.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authorizationv1.ResourceAttributes{
+				Namespace:   namespace,
+				Verb:        string(app.Verb),
+				Group:       gvr.Group,
+				Resource:    gvr.Resource,
+				Name:        name,
+				Subresource: app.Subresource,
+			},
+		},
+	}
+	response, err := tmpAuthClient.SelfSubjectAccessReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	if response.Status.Allowed {
+		return nil
+	}
+	var errMsg = fmt.Sprintf("resource %v authorize failed.", gvr)
+	if len(response.Status.Reason) > 0 {
+		errMsg += fmt.Sprintf("reason: %v.", response.Status.Reason)
+	}
+	if len(response.Status.EvaluationError) > 0 {
+		errMsg += fmt.Sprintf("evaluation error: %v.", response.Status.EvaluationError)
+	}
+	return fmt.Errorf(errMsg)
+}
+
 // ProcessApplication processes application by re-translating it to kube-api request with kube client,
 // which will be processed and responded by apiserver eventually.
 // Specially if app.verb == watch, it transforms app to a listener and register it to HandlerCenter, rather
 // than request to apiserver directly. Listener will then continuously listen kube-api change events and
 // push them to edge node.
 func (c *Center) ProcessApplication(app *Application) (interface{}, error) {
 	app.Status = InProcessing
-
 	gvr, ns, name := metaserver.ParseKey(app.Key)
+	err := c.authorizeApplication(app, gvr, ns, name)
```

Comment:
I think some optimizations can be made. For short connection requests, such as delete、update、and so on, we can directly use the constructed client to request, otherwise we need to request for every request.

Reply:
Done.